store/tikv/txn.go

```diff
@@ -195,8 +195,12 @@ func (txn *tikvTxn) Commit(ctx context.Context) error {
 	}
 
 	// latches enabled
-	// for transactions not retryable, commit directly.
-	if !sessionctx.GetRetryable(ctx) {
+	var forUpdate bool
+	if option := txn.us.GetOption(kv.ForUpdate); option != nil {
```

Comment:
The ForUpdate option is set directly on txn, why we get it from txn.us?

> s.txn.SetOption(kv.ForUpdate, true)

Reply:
Because  the definition of Transaction:

```
type Transaction interface {
	SetOption(opt Option, val interface{})
	// DelOption deletes an option.
	DelOption(opt Option)
        ...
}
```

It doesn't have a `GetOption` function !
So we can 

```
txn.SetOption(xxx)
```

but we can't

```
txn.GetOption(xxx)
```

Reply:
Maybe we could implement `GetOption` for `Transaction` in the future?